app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/CellPlotService.java

```diff
@@ -103,27 +103,30 @@ public ImmutableSet<TSnePoint> expressionPlot(String experimentAccession,
     }
 
     public ImmutableMap<String, JsonObject> fetchDefaultPlotMethodWithParameterisation(String experimentAccession) {
-        var umap = "umap";
-        var tsne = "tsne";
+
         var defaultcellPlots = cellPlotDao.fetchDefaultPlotMethodWithParameterisation(experimentAccession);
 
         ImmutableMap.Builder<String, JsonObject> defaultPlotTypeAndOptions = new ImmutableMap.Builder<>();
-        if (!defaultcellPlots.isEmpty()) {
-            defaultPlotTypeAndOptions.put(umap, getMiddleElement(defaultcellPlots.get(umap)));
-            defaultPlotTypeAndOptions.put(tsne, getMiddleElement(defaultcellPlots.get(tsne)));
-        }
+
+        defaultcellPlots.forEach((method, options) -> defaultPlotTypeAndOptions.put(method, getMiddleElement(defaultcellPlots.get(method))));
+
         return defaultPlotTypeAndOptions.build();
     }
 
-    private static JsonObject getMiddleElement(List plotOptions) {
-
-        if (plotOptions.size() % 2 == 0) { // even number
-            Object umapEvenItem = plotOptions.get((plotOptions.size() / 2 - 1));
-            return (JsonObject) umapEvenItem;
-        } else { //odd number
-            Object umapOddItem = plotOptions.get((plotOptions.size() / 2));
-            return (JsonObject) umapOddItem;
+    private JsonObject getMiddleElement(List plotOptions) {
+
+        if (plotOptions.isEmpty() || plotOptions == null) {
+            return new JsonObject();
```

Comment:
Is an empty `JsonObject` is a valid return value here?
While I am not sure about the answer I won't check the tests.
That's one of the reason that it is better to write a test/tests first, but it looks like you wrote the tests later. Am I correct?

Reply:
I fixed your comments @ke4. Yes, correct,  I fixed the tests after implementation. Now, I managed to assert an Empty JSON string.

I'm also not happy with that empty JSON object :).

How about throwing a custom exception when we met this condition `if (plotOptions == null || plotOptions.isEmpty()) {`

Reply:
I think it is probably better to throw an exception, but we also need to think if that could happen at all OR if with a refactor we could eliminate this case.